modules/client/src/main/java/org/apache/ignite/internal/client/table/AbstractClientView.java

```diff
@@ -70,9 +112,46 @@ protected static SqlSerializer createSqlSerializer(String tableName, ClientColum
                 .build();
     }
 
+    /**
+     * Create conversion function for objects contained by result set to criteria query objects.
+     *
+     * @param meta Result set columns' metadata.
+     * @param schema Schema.
+     * @return Conversion function (if {@code null} conversions isn't required).
+     */
+    protected @Nullable Function<SqlRow, T> queryMapper(ResultSetMetadata meta, ClientSchema schema) {
+        return null;
+    }
+
     /** {@inheritDoc} */
     @Override
     public Cursor<T> query(@Nullable Transaction tx, @Nullable Criteria criteria, @Nullable CriteriaQueryOptions opts) {
         return new CursorAdapter<>(sync(queryAsync(tx, criteria, opts)));
     }
+
+    /** {@inheritDoc} */
+    @Override
+    public CompletableFuture<AsyncCursor<T>> queryAsync(
+            @Nullable Transaction tx,
+            @Nullable Criteria criteria,
+            CriteriaQueryOptions opts
```

Comment:
1. Can `opts` be null? Either add a null check, or `@Nullable`
2. `opts` in unused?

Reply:
Fixed